[CMake] All post-commit buildbots should build with --no-fatal-warnings

mcatanzaro opened the pull request:
#### e868bfe3d84531d77d21eecc5f67c27aee1bb503
<pre>
[CMake] All post-commit buildbots should build with --no-fatal-warnings
<a href="https://bugs.webkit.org/show_bug.cgi?id=246760">https://bugs.webkit.org/show_bug.cgi?id=246760</a>

Reviewed by Jonathan Bedard.

We need to enable fatal warnings on EWS to prevent developers from
introducing new warnings. But we do not want fatal warnings on
post-commit bots, since that may result in lost test results, which
would be a very harsh price to pay for an unused variable or whatnot.

Note this is not needed for the JSC bots since those do not use
ENABLE_DEVELOPER_MODE.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success):
(TestCompileWebKit.test_success_gtk):
(TestCompileWebKit.test_success_wpe):
(TestCompileWebKit.test_failure):

Canonical link: <a href="https://commits.webkit.org/255954@main">https://commits.webkit.org/255954@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6c7213459a93494fc2da0fb24dd9bd9523841b1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94070 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3260 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24613 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103695 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164039 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98061 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3276 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31484 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86377 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99739 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99729 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2354 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80491 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29365 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84280 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72317 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37872 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17788 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/93078 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35749 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19054 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39623 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/41628 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1935 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41564 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38286 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
These changes will need to be landed either squashed or in two separate PRs, we'll also need to coordinate a build.webkit.org restart.

lauromoura commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -411,224 +411,252 @@
                     {
                       "name": "GTK-Linux-64-bit-Release-Tests", "factory": "TestAllButJSCFactory",
                       "platform": "gtk", "configuration": "release", "architectures": ["x86_64"],
+                      "additionalArguments": ["--no-fatal-warnings"],
```
Won't this break the test steps? `run-webkit-tests` complains when given this option, likely other test scripts too.

Maybe instead of changing the config file we could concentrate this on the `CompileWebKit` step definition here? https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L2476

mcatanzaro replied:
Hm, OK, but is the ews-build code used by the post-commit buildbots? That's not just for EWS alone, is it?

mcatanzaro replied:
Um, if that steps.py is used for both EWS and post-commit bots... any idea how to tell the difference? Goal is to have fatal warnings on all the EWS bots, but not on any of the post-commit bots.

lauromoura replied:
Oh, sorry. I linked the wrong file. For post commit it's this one: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/build-webkit-org/steps.py#L348

mcatanzaro replied:
Yay, great. I'll try to implement your suggestion.

mcatanzaro committed:
[CMake] All post-commit buildbots should build with --no-fatal-warnings
https://bugs.webkit.org/show_bug.cgi?id=246760

Reviewed by Jonathan Bedard.

We need to enable fatal warnings on EWS to prevent developers from
introducing new warnings. But we do not want fatal warnings on
post-commit bots, since that may result in lost test results, which
would be a very harsh price to pay for an unused variable or whatnot.

Note this is not needed for the JSC bots since those do not use
ENABLE_DEVELOPER_MODE.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success):
(TestCompileWebKit.test_success_gtk):
(TestCompileWebKit.test_success_wpe):
(TestCompileWebKit.test_failure):

Canonical link: https://commits.webkit.org/255954@main
